backend/core.go

```diff
@@ -8,14 +8,21 @@ import (
 	"github.com/grafana/grafana-plugin-sdk-go/dataframe"
 )
 
-// PluginConfig holds configuration for the queried plugin.
+// PluginConfig configuration for a plugin.
```

Comment:
In regards to naming. Shouldn't we make PluginConfig the most generic config and only carry values that all plugins need. Ex `PluginId`, `Type`. It's a bit confusing for me that PluginConfig carries all datasource settings.